.github/workflows/ChecksAndLinters.yml

```diff
@@ -0,0 +1,72 @@
+name: checkers_and_linters
+
+# Controls when the workflow will run
+on:
+  # Triggers the workflow on push or pull request events but only for some branches
+  push:
+    branches: [ main, angi ]
+  pull_request:
+    branches: [ main, stable, angi ]
+
+jobs:
+  # This workflow contains a single job called "checksAndlinters"
+  ChecksAndLinters:
+    # The type of runner that the job will run on
+    runs-on: ubuntu-latest
+
+    # Steps represent a sequence of tasks that will be executed as part of the job
+    steps:
+      # Checks-out your repository under $GITHUB_WORKSPACE, so your job can access it
+      - uses: actions/checkout@v2
+
+      - name: Install needed packages
+        run: |
+          echo "Perl Linter"
+          echo "Install perlcritic..."
+          id -a
+          sudo apt-get -y install libtest-perl-critic-perl
+          which perlcritic
+          echo "Flake8"
+          echo "Install cke8..."
+          id -a
+          sudo apt-get -y install flake8
+          which flake8
+          echo "Pylint"
+          echo "Install pylint..."
+          id -a
+          sudo apt-get -y install pylint
+          which pylint
+        continue-on-error: true
+
+      # Runs a set of commands using the runners shell
+      - name: Checks scripts with shellcheck
+        shell: bash
+        run: |
+          echo "Running Shellcheck..."
+          find . -type f  -name '*' ! -path './ra/old-*' ! -path './test/*' -exec awk ' /^#!.*bash/{print FILENAME} {nextfile}' {} + | xargs shellcheck -s bash && echo "Everything is OK!"
```

Comment:
@angelabriel I am not sure, if I am right here. But I guess `shellcheck -s bash <file>` will fail for SAPHanaController and SAPHanaTopology. I would prefer `shellcheck -s bash -a -x <file>`. This tells shellcheck to also include the shell libraries. This is e.g. needed to check for the undefined variables.